Jenkinsfile

```diff
@@ -140,12 +141,27 @@ pipeline {
           sh 'helmfile version'
 
           echo 'synchronize cluster state'
+          helmfileTemplate("${HELMFILE_ENVIRONMENT}", 'target')
           withCredentials([
             usernamePassword(credentialsId: 'packages.nuxeo.com-auth', usernameVariable: 'PACKAGES_USERNAME', passwordVariable: 'PACKAGES_PASSWORD'),
             usernamePassword(credentialsId: 'connect-prod', usernameVariable: 'CONNECT_USERNAME', passwordVariable: 'CONNECT_PASSWORD'),
           ]) {
-            helmfileTemplate("${HELMFILE_ENVIRONMENT}", 'target')
-            helmfileSync("${HELMFILE_ENVIRONMENT}")
+            script {
+              def awsAccessKeyId = sh(
+                script: "kubectl get secret ${AWS_CREDENTIALS_SECRET} -n ${NAMESPACE} -o=jsonpath='{.data.access_key_id}' | base64 --decode",
+                returnStdout: true
+              )
+              def awsSecretAccessKey = sh(
+                script: "kubectl get secret ${AWS_CREDENTIALS_SECRET} -n ${NAMESPACE} -o=jsonpath='{.data.secret_access_key}' | base64 --decode",
+                returnStdout: true
+              )
+              withEnv([
+                  "AWS_ACCESS_KEY_ID=${awsAccessKeyId}",
+                  "AWS_SECRET_ACCESS_KEY=${awsSecretAccessKey}"
+                ]) {
```

Comment:
I think we can do the same thing as `packages.nuxeo.com-auth` and `connect-prod`, and then use`usernamePassword` directive to populate the environment.

To allow the key mapping the secret should be labelled and annotated accordingly, for example from the [jenkins credential plugin documentation](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/examples/#custom-field-mapping):
```yaml
apiVersion: v1
kind: Secret
metadata:
# this is the jenkins id.
  name: "a-test-usernamepass"
  labels:
    # so we know what type it is.
    "jenkins.io/credentials-type": "usernamePassword"
  annotations:
    # description - can not be a label as spaces are not allowed
    "jenkins.io/credentials-description" : "credentials from Kubernetes"
    # map the username field to user
    "jenkins.io/credentials-keybinding-username" : "user"
    # map the password field to pass
    "jenkins.io/credentials-keybinding-password" : "pass"
type: Opaque
stringData:
  user: myUsername
  pass: 'Pa$$word'
```

What do you think?

Reply:
Indeed, that was our first approach, but it displays the AWS access key in the Jenkins credentials screen, as if it was a username (`AWS_KEY_VALUE/*******`).
Though the access key isn't considered as a secret, and this screen is reserved for admins, I would rather not display it